betterproto/plugin.py

```diff
@@ -30,38 +30,55 @@
 
 from betterproto.casing import safe_snake_case
 
+import google.protobuf.wrappers_pb2
+
 
 WRAPPER_TYPES = {
-    "google.protobuf.DoubleValue": "float",
-    "google.protobuf.FloatValue": "float",
-    "google.protobuf.Int64Value": "int",
-    "google.protobuf.UInt64Value": "int",
-    "google.protobuf.Int32Value": "int",
-    "google.protobuf.UInt32Value": "int",
-    "google.protobuf.BoolValue": "bool",
-    "google.protobuf.StringValue": "str",
-    "google.protobuf.BytesValue": "bytes",
+    google.protobuf.wrappers_pb2.DoubleValue: "float",
+    google.protobuf.wrappers_pb2.FloatValue: "float",
+    google.protobuf.wrappers_pb2.Int64Value: "int",
+    google.protobuf.wrappers_pb2.UInt64Value: "int",
+    google.protobuf.wrappers_pb2.Int32Value: "int",
+    google.protobuf.wrappers_pb2.UInt32Value: "int",
+    google.protobuf.wrappers_pb2.BoolValue: "bool",
+    google.protobuf.wrappers_pb2.StringValue: "str",
+    google.protobuf.wrappers_pb2.BytesValue: "bytes",
 }
 
 
-def get_ref_type(package: str, imports: set, type_name: str) -> str:
+def get_wrapper_type(type_name: str) -> (Any, str):
```

Comment:
I'm not sure about the trade off here in keeping WRAPPER_TYPES declarative as a simple mapping at the cost of not being able to actually use it directly as a mapping.

What about replacing WRAPPER_TYPES and get_wrapper_type here with:

```python
WRAPPER_TYPES = {
    goog_type.DESCRIPTOR.full_name: (goog_type, simple_name)
    for goog_type, simple_name in (
        (google.protobuf.wrappers_pb2.DoubleValue, "float"),
        (google.protobuf.wrappers_pb2.FloatValue, "float"),
        (google.protobuf.wrappers_pb2.Int64Value, "int"),
        (google.protobuf.wrappers_pb2.UInt64Value, "int"),
        (google.protobuf.wrappers_pb2.Int32Value, "int"),
        (google.protobuf.wrappers_pb2.UInt32Value, "int"),
        (google.protobuf.wrappers_pb2.BoolValue, "bool"),
        (google.protobuf.wrappers_pb2.StringValue, "str"),
        (google.protobuf.wrappers_pb2.BytesValue, "bytes"),
    )
}
```

Reply:
I see what you mean, good suggestion! That `get_wrapper_type ` a bit ugly indeed, and would be nice to not need it, while at the same time, I did not like the unchecked references to packages. Your idea solves both, although it embarrassingly took me a several re-reads to fully understand it.

To build on your idea, how about two maps? Same concept but less inlining..

```python
WRAPPER_TYPES = {
    google.protobuf.wrappers_pb2.DoubleValue: "float",
    google.protobuf.wrappers_pb2.FloatValue: "float",
    google.protobuf.wrappers_pb2.Int64Value: "int",
    google.protobuf.wrappers_pb2.UInt64Value: "int",
    google.protobuf.wrappers_pb2.Int32Value: "int",
    google.protobuf.wrappers_pb2.UInt32Value: "int",
    google.protobuf.wrappers_pb2.BoolValue: "bool",
    google.protobuf.wrappers_pb2.StringValue: "str",
    google.protobuf.wrappers_pb2.BytesValue: "bytes",
}

WRAPPERS = {w.DESCRIPTOR.full_name: w for w in WRAPPER_TYPES.keys()}

# usage
wrapper = WRAPPERS['google.protobuf.UInt64Value']

print(wrapper.__module__)
print(WRAPPER_TYPES[wrapper])
```

the only think i don't like about it is having two similar maps with similar names.

Reply:
My preference is for the single WRAPPER_TYPES dict for conciseness, though I get that the idiom might not be as clear.

As you prefer.

Reply:
Well, I found out we could also just extract the desired type from instance().value , so we don't need the tuples in the map

```python
wrappers = {
    'google.protobuf.DoubleValue': google.protobuf.wrappers_pb2.DoubleValue,
    'google.protobuf.FloatValue' : google.protobuf.wrappers_pb2.FloatValue,
    'google.protobuf.Int64Value' : google.protobuf.wrappers_pb2.Int64Value,
    'google.protobuf.UInt64Value' : google.protobuf.wrappers_pb2.UInt64Value,
    'google.protobuf.Int32Value' : google.protobuf.wrappers_pb2.Int32Value,
    'google.protobuf.UInt32Value' : google.protobuf.wrappers_pb2.UInt32Value,
    'google.protobuf.BoolValue' : google.protobuf.wrappers_pb2.BoolValue,
    'google.protobuf.StringValue' : google.protobuf.wrappers_pb2.StringValue,
    'google.protobuf.BytesValue' : google.protobuf.wrappers_pb2.BytesValue,
}
```

Reply:
I've implemented this concept. Please see #8f0caf1